### PR TITLE
Adds new microtransaction method: Scratch Cards!

### PR DESCRIPTION
**Pull request by Henri215**
## What Does This PR Do

Adds scratch cards, available from mails (box with 5), merchandise vendor (box with 5 - 100 credits) and maints loot.
You can scratch it with either a card or coin to reveal if you won or not (1% chance)!
Redeem your prize (1000 credits) by using the scratch card on ATMs!

Scratch sound from https://freesound.org/people/Henaway/sounds/127979/, CC0.

## Why It's Good For The Game

This is an alternative to the slot machines if someone wants to ~~get poor~~ try their luck and it adds a cool usage to IDs and coins.
## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/77684085/c814580c-86d2-4025-963c-915cc3409788

## Testing

- See video above, did also acquire it from the vendor, maints and mail.
## Changelog
:cl: Henri215, mcramon
add: Added scratch cards, available from mails, merchandise vendors and maints loot
/:cl:

### CODE REVIEW

File: code/game/objects/items/scratch_card.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			icon_state = "scard_loser"
+		playsound(user, 'sound/items/scratching.ogg', 50, TRUE)
+		scratched = TRUE
+		update_icon_state()
+
+/obj/item/scratch/attack_obj(obj/O, mob/living/user, params)
+	if(winner && istype(O, /obj/machinery/economy/atm))
+		playsound(user, 'sound/machines/ping.ogg', 50, TRUE)
+		O.atom_say("Congratulations for winning the lottery!")
+		var/obj/item/reward = new /obj/item/stack/spacecash/c1000(get_turf(src))
```
Comment (DGamerL) [inline code/game/objects/items/scratch_card.dm:34]:
```suggestion
		var/obj/item/reward = new /obj/item/stack/spacecash/c1000
```
I don't think we need the location anyways, all we need is a reference to put into their hands

File: code/modules/economy/merch_items.dm [hunk 2]
```diff
@@ -47,6 +47,13 @@
 	cost = 125
 	category = MERCH_CAT_TOY
 
+/datum/merch_item/scratch_card
+	name = "Scratch Cards Box"
+	desc = "Contains 5 scratch cards, become rich today!"
+	typepath = /obj/item/storage/box/scratch_cards
+	cost = 150
```
Comment (DGamerL) [inline code/modules/economy/merch_items.dm:54]: Might be a little too high in this economy?

File: code/modules/economy/merch_items.dm [hunk 3]
```diff
@@ -47,6 +47,13 @@
 	cost = 125
 	category = MERCH_CAT_TOY
 
+/datum/merch_item/scratch_card
+	name = "Scratch Cards Box"
+	desc = "Contains 5 scratch cards, become rich today!"
```
Comment (DGamerL) [inline code/modules/economy/merch_items.dm:52]:
```suggestion
	desc = "Contains 5 scratch cards. Become rich today!"
```

**Comment by McRamon** [comment]
would be good if each time you scratched one, it would give you a message in chat (the  "good luck next time" and " you won, go to atm")

**Comment by Henri215** [comment]
> would be good if each time you scratched one, it would give you a message in chat (the "good luck next time" and " you won, go to atm")

added

**Comment by ATP-Engineer** [comment]
Maybe two different scratch cards, ones for the big lottery. And getrichquickTM scratchers that have a supstancially higher win chance at a lower win reward. IE, higher cost to buy, but 19% more likely to win like 200-400 credits.

**Comment by Landerlow** [comment]
Imagine saying this as a joke/half baked idea in LOOC and Henri hears you. And not even a day later this PR is up. 
Amazing work :D Can't wait to scratch all the cards.

**Comment by Henri215** [comment]
Updated with new sprites made by mcramon
image

File: code/game/objects/items/scratch_card.dm [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		scratched = TRUE
+		update_icon_state()
+
+/obj/item/scratch/attack_obj(obj/O, mob/living/user, params)
+	if(winner && istype(O, /obj/machinery/economy/atm))
+		playsound(user, 'sound/machines/ping.ogg', 50, TRUE)
+		O.atom_say("Congratulations for winning the lottery!")
+		var/obj/item/reward = new /obj/item/stack/spacecash/c1000
+		user.put_in_hands(reward)
+		qdel(src)
```
Comment (DGamerL) [inline code/game/objects/items/scratch_card.dm:36]: We might want to qdel the card before we put it in their hands, so it'll go into the hand with the card!

File: code/game/objects/items/scratch_card.dm [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+		return
+	..()
+
+/obj/item/storage/box/scratch_cards
+	name = "scratch cards box"
+	desc = "Try your luck with five scratch cards!"
+	can_hold = list(/obj/item/scratch)
```
Comment (DGamerL) [inline code/game/objects/items/scratch_card.dm:43]: Could you make this a weight class smaller just so it'd be more fitting to be a mail item?

Comment (Henri215) [reply code/game/objects/items/scratch_card.dm:43]: Done

File: code/game/objects/items/scratch_card.dm (hunk 4 above)
Comment (Henri215) [reply code/game/objects/items/scratch_card.dm:36]: Done

**Comment by SteelSlayer** [comment]
Scratch sound is a bit loud imo, but otherwise good.

**Comment by Henri215** [comment]
> Scratch sound is a bit loud imo, but otherwise good.

Oh, i decreased it without seeing your comment, will probably make another video later with the new sprites.

**Comment by Henri215** [comment]
Updated the video

**Review by Burzah — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**